lib/src/over_react_test/common_component_util.dart

```diff
@@ -406,25 +409,67 @@ void testRequiredProps(BuilderOnlyUiFactory factory, dynamic childrenFactory())
   });
 
   test('throws when the required prop is not set or is null', () {
-    requiredProps.forEach((String propKey) {
-      final reactComponentFactory = factory().componentFactory as ReactDartComponentFactoryProxy; // ignore: avoid_as
+    if (!useComponent2Tests) {
+      requiredProps.forEach((String propKey) {
+        final reactComponentFactory = factory().componentFactory as ReactDartComponentFactoryProxy; // ignore: avoid_as
 
-      // Props that are defined in the default props map will never not be set.
-      if (!reactComponentFactory.defaultProps.containsKey(propKey)) {
+        // Props that are defined in the default props map will never not be set.
+        if (!reactComponentFactory.defaultProps.containsKey(propKey)) {
+          var badRenderer = () => render((factory()
+            ..remove(propKey)
+          )(childrenFactory()));
+
+          expect(badRenderer, throwsPropError_Required(propKey, keyToErrorMessage[propKey]), reason: '$propKey is not set');
+        }
+
+        var propsToAdd = {propKey: null};
         var badRenderer = () => render((factory()
-          ..remove(propKey)
+          ..addAll(propsToAdd)
         )(childrenFactory()));
 
-        expect(badRenderer, throwsPropError_Required(propKey, keyToErrorMessage[propKey]), reason: '$propKey is not set');
-      }
+        expect(badRenderer, throwsPropError_Required(propKey, keyToErrorMessage[propKey]), reason: '$propKey is set to null');
+      });
+    } else {
+      PropTypes.resetWarningCache();
+
+      List<String> consoleErrors = [];
+      JsFunction originalConsoleError = context['console']['error'];
+      context['console']['error'] = new JsFunction.withThis((self, [message, arg1, arg2, arg3,  arg4, arg5]) {
+        consoleErrors.add(message);
+        originalConsoleError.apply([message], thisArg: self);
+      });
 
-      var propsToAdd = {propKey: null};
-      var badRenderer = () => render((factory()
-        ..addAll(propsToAdd)
-      )(childrenFactory()));
+      final reactComponentFactory = factory().componentFactory as
+          ReactDartComponentFactoryProxy2; // ignore: avoid_as
 
-      expect(badRenderer, throwsPropError_Required(propKey, keyToErrorMessage[propKey]), reason: '$propKey is set to null');
-    });
+      requiredProps.forEach((String propKey) {
+        if (!reactComponentFactory.defaultProps.containsKey(propKey)) {
+
+          mount((factory()
+            ..remove(propKey)
+          )(childrenFactory()));
+
+          expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
+          expect(consoleErrors, [contains(keyToErrorMessage[propKey])],
+              reason: '$propKey is not set');
+        }
+
+        context['console']['error'] = originalConsoleError;
```

Comment:
Your currently getting false positives here, by setting this back to original you wouldnt catch any new errors, and so in the next set of expects, its actually passing because its the old error from before so we should instead clear the consoleErrors here. and because propTypes doesnt throw the same error more than once per session we should also have `PropTypes.resetWarningCache();` in there to reset the cache so it will error again.
```suggestion
        consoleErrors = [];
        PropTypes.resetWarningCache();
```

Reply:
Ahhh, yep, that makes sense. Thank you, Keal!